go/src/koding/kites/config/cache.go

```diff
@@ -34,19 +35,39 @@ type Cache struct {
 // If it was not possible to create or open BoltDB database,
 // an in-memory cache is created.
 func NewCache(options *CacheOptions) *Cache {
+	db, err := newBoltDB(options)
+	if err != nil {
+		log.Println(err)
+	}
+
 	return &Cache{
-		EncodingStorage: storage.NewEncodingStorage(newBoltDB(options), options.Bucket),
+		EncodingStorage: storage.NewEncodingStorage(db, options.Bucket),
 	}
 }
 
-func newBoltDB(o *CacheOptions) *bolt.DB {
-	os.MkdirAll(filepath.Dir(o.File), 0755)
+// NewCache returns new cache value backed by BoltDB.
+func NewBoltCache(options *CacheOptions) (*Cache, error) {
+	db, err := newBoltDB(options)
+	if err != nil {
+		return nil, err
+	}
 
-	if db, err := bolt.Open(o.File, 0644, o.BoltDB); err == nil {
-		return db
+	bolt, err := storage.NewBoltStorageBucket(db, options.Bucket)
+	if err != nil {
+		return nil, err
 	}
 
-	return nil
+	return &Cache{
+		EncodingStorage: &storage.EncodingStorage{
+			Interface: bolt,
+		},
+	}, nil
+}
+
+func newBoltDB(o *CacheOptions) (*bolt.DB, error) {
+	os.MkdirAll(filepath.Dir(o.File), 0755)
```

Comment:
stating "minor" again, i just mentioned the error message, `no such file or directory` message does not say anything about why that file/folder is required or anything